tests/System.Slices.Tests/MemoryTests.cs

```diff
@@ -30,25 +31,72 @@ public void SimpleTestS()
         [Fact]
         public void ArrayMemoryLifetime()
         {
-            Memory<byte> copyStoredForLater;
-            using (var owner = new OwnedArray<byte>(1024)) {
-                Memory<byte> memory = owner.Memory;
-                Memory<byte> memorySlice = memory.Slice(10);
-                copyStoredForLater = memorySlice;
-                using (memorySlice.Reserve()) { // increments the "outstanding span" refcount
-                    Assert.Throws<InvalidOperationException>(() => { // memory is reserved; cannot dispose
-                        owner.Dispose();
-                    });
-                    Span<byte> span = memorySlice.Span;
-                    span[0] = 255;
-                } // releases the refcount
-            }
+            Memory<byte> copyStoredForLater = default(Memory<byte>);
+            Assert.Throws<InvalidOperationException>(() => {
```

Comment:
Dispose true/false already has meaning. I don't want to hijack it for something else. On the other hand, we have many types in the framework that allow multiple dispose calls. The subsequent ones are just nops. But in this case, maybe we want to detect them (without failing)

@davidfowl, @jkotas?